File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    const col = arr.length;
    const row = arr[0].length;
    
    if(col === row) return arr;
    if(col > row) {
        return arr.map((l) => [...l, ...Array(col-row).fill(0)]);
    } else {
        const newArr = Array(row-col).fill().map(() => Array(row).fill(0))
        return [...arr, ...newArr];
    }
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int row = arr.length;
        int col = arr[0].length;
        
        int size = Math.max(row, col);
        int[][] result = new int[size][size];
        
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                result[i][j] = arr[i][j];
            }
        }
        
        return result;
    }
}